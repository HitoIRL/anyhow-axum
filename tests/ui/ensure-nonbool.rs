use anyhow::{ensure, Result};

fn main() -> Result<()> {
    ensure!("...");

    struct Struct(bool);
    let mut s = Struct(true);
    match &mut s {
        Struct(cond) => ensure!(cond),
    }

    Ok(())
}
