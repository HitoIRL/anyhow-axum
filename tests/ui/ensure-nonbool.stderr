error[E0600]: cannot apply unary operator `!` to type `&'static str`
 --> tests/ui/ensure-nonbool.rs:4:5
  |
4 |     ensure!("...");
  |     ^^^^^^^^^^^^^^ cannot apply unary operator `!`
  |
  = note: this error originates in the macro `$crate::__fallback_ensure` which comes from the expansion of the macro `ensure` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0600]: cannot apply unary operator `!` to type `&mut bool`
 --> tests/ui/ensure-nonbool.rs:9:25
  |
9 |         Struct(cond) => ensure!(cond),
  |                         ^^^^^^^^^^^^^ cannot apply unary operator `!`
  |
  = note: this error originates in the macro `$crate::__fallback_ensure` which comes from the expansion of the macro `ensure` (in Nightly builds, run with -Z macro-backtrace for more info)
